pyquil/api/_qpu.py

```diff
@@ -142,6 +144,27 @@ def _build_patch_values(self) -> dict:
                 continue
             patch_table[name] = [0] * spec.length
 
+        # Write in the values for arithmetic parameter expressions in the original program
+        # For example:
+        #     DECLARE theta REAL
+        #     RZ(3 * theta) 0
+        # gets translated to:
+        #     DECLARE theta REAL
+        #     DECLARE __P REAL[0]
+        #     RZ(__P[0]) 0
+        # and the recalculation table gets the entry: __P[0] = '3 * theta'
+        for memory_reference, recalculation_rule in self._executable.recalculation_table.items():
+            # Parse the expression given by the recalculation rule
+            # TODO: After #687, we should be able to parse this in a better way
+            expression = parser.parse(f"RZ({recalculation_rule}) 0")[0].params[0]
```

Comment:
if this is slow, it should probably be done during `load`

Reply:
great idea! Even if it's not slow I prefer it